core/store/src/main/java/mil/nga/giat/geowave/core/store/DataStore.java

```diff
@@ -267,6 +268,29 @@ public boolean deleteEntry(
 
 	/**
 	 * Returns all data in this data store that matches the query parameter
+	 * within the index described by the index passed in. All data types that
+	 * match the query will be returned as an instance of the native data type
+	 * that was originally ingested, but only the subset of attributes specified
+	 * will be included.
+	 * 
+	 * @param index
+	 *            The index information to query against. All data within the
+	 *            index of this index ID will be queried and returned.
+	 * @param attributes
+	 *            The subset of attributes to include in the results
+	 * @param query
+	 *            The description of the query to be performed
+	 * @return An iterator on all results that match the query. The iterator
+	 *         implements Closeable and it is best practice to close the
+	 *         iterator after it is no longer needed.
+	 */
+	public <T> CloseableIterator<T> query(
```

Comment:
hmm, at first, the query methods were intended to be intuitive and expose things that are sort of expected in general with query (such as the common SQL syntax).  So 'limit' was a parameter rather than hidden in an object...

Now we're getting to the point of too many options and therefore too many query methods...I agree with @chrisbennight that it may make sense to have a set of concrete query options that could apply to any query and then have a separate interface/abstraction that handles the query body which would change based on the type of index (spatial, spatial-temporal, or whatever else).  The question is would we rather have QueryOptions as a parameter or have a query object that contains the concrete set of options and the query body?  The only argument that would make me lean to the latter is that often no limit or field subsetting are used, so you could instantiate the query object with the query body and the default options would not need to be specified, but if it was part of the method signature you'd have to instantiate some sort of query options or pass null.  Not a big deal, but the datastore is pretty much at the core of everything.  Thoughts?